api/cluster/resource/templater.go

```diff
@@ -385,13 +390,26 @@ func (t *InferenceServiceTemplater) enrichStandardTransformerEnvVars(envVars mod
 	addEnvVar := models.EnvVars{}
 	addEnvVar = append(addEnvVar, models.EnvVar{Name: transformerpkg.DefaultFeastSource, Value: t.standardTransformerConfig.DefaultFeastSource.String()})
 
-	// adding feast storage config env variable
+	// adding feast related config env variable
 	feastStorageConfig := t.standardTransformerConfig.ToFeastStorageConfigs()
-
 	if feastStorageConfigJsonByte, err := json.Marshal(feastStorageConfig); err == nil {
 		addEnvVar = append(addEnvVar, models.EnvVar{Name: transformerpkg.FeastStorageConfigs, Value: string(feastStorageConfigJsonByte)})
 	}
 
+	// Add keepalive configuration for predictor
+	// only pyfunc config that enforced by merlin
+	keepAliveModelCfg := t.standardTransformerConfig.ModelClientKeepAlive
+	if modelService.Protocol == protocol.UpiV1 && modelService.Type == models.ModelTypePyFunc {
+		addEnvVar = append(addEnvVar, models.EnvVar{Name: transformerpkg.ModelGRPCKeepAliveEnabled, Value: strconv.FormatBool(keepAliveModelCfg.Enabled)})
```

Comment:
With this code, it would cause a duplicate environment variable if `addEnvVar` contains the same environment variable. Can we make sure that merlin only add these environment variable if users doesn't specify one?

Reply:
that won't happen since we do merge `envVars = models.MergeEnvVars(addEnvVar, envVars)` if there is same key it will use value from `envVars`

Reply:
Ah ok make sense